### PR TITLE
[feat]: [DBOPS-307]: Onboard DBDevOps harness_platform_db_instance

### PR DESCRIPTION
**Pull request by Abhishek-CDC**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: internal/service/platform/db_instance/data_source_db_instance.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+func DataSourceDBInstance() *schema.Resource {
+	resource := &schema.Resource{
+		Description: "Data source for retrieving a Harness DBDevOps Instance.",
+
+		ReadContext: dataSourceDBInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"schema": {
+				Description: "The reference to schema",
```
Comment (yogesh-chauhan) [inline internal/service/platform/db_instance/data_source_db_instance.go:23]: Lets call it "The identifier of the parent database schema"

File: internal/service/platform/db_instance/resource_db_instance.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		Schema: map[string]*schema.Schema{
+			"schema": {
+				Description: "The reference to schema",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"branch": {
+				Description: "The branch of changeSet repository",
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (yogesh-chauhan) [inline internal/service/platform/db_instance/resource_db_instance.go:34]: branch will not be required for artifactory connector

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: internal/service/platform/db_instance/resource_db_instance.go (hunk 2 above)
Comment (Abhishek-CDC) [reply internal/service/platform/db_instance/resource_db_instance.go]: Done

File: internal/service/platform/db_instance/data_source_db_instance.go (hunk 1 above)
Comment (Abhishek-CDC) [reply internal/service/platform/db_instance/data_source_db_instance.go]: Done

**Review by yogesh-chauhan — APPROVED** [review]

**Pull request merged**